### PR TITLE
[RAP-46] Add /domains endpoint to reseller API specs

### PR DESCRIPTION
**Pull request by Hrnt125**
Add description for 'getMultiple' endpoint of Reseller API.
TODO: possibly need to fix 401 response for /domains/{domainName}.

### CODE REVIEW

File: openapi.yaml [hunk 1]
```diff
@@ -377,6 +412,12 @@ components:
           type: string
           example: 'dao'
   responses:
+    ResellerNotFound:
```
Comment (fruwe) [inline openapi.yaml:415]: Let's use `PartnerNotFound` instead

File: openapi.yaml [hunk 2]
```diff
@@ -632,35 +679,46 @@ components:
...
-              type: boolean
-              example: true
+          $ref: '#/components/schemas/Availability'
+    DomainsAvailabilityResponse:
+      type: object
+      properties:
+        domains:
+          type: array
+          items:
+            $ref: '#/components/schemas/DomainAvailabilityResponse'
```
Comment (fruwe) [inline openapi.yaml:720]:
```suggestion
            $ref: '#/components/schemas/Availability'
```

File: openapi.yaml [hunk 3]
```diff
@@ -31,6 +31,25 @@ tags:
...
+      summary: Multiple domain names availability
+      responses:
+        '200':
+          description: Successful
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/DomainsAvailabilityResponse'
+        '400':
+          $ref: '#/components/responses/DomainsBadRequest'
```
Comment (fruwe) [inline openapi.yaml:50]: I think we can use `#/components/responses/BadRequest` should be good enough.

**Review by fruwe — CHANGES_REQUESTED** [review]

File: openapi.yaml (hunk 2 above)
Comment (Hrnt125) [reply openapi.yaml:720]: { "domains": [
   {
     # same as return value of /domains/{domainName} 
   }
] }

Availability is just 1 out of 2 parts we return in response.

File: openapi.yaml (hunk 1 above)
Comment (Hrnt125) [reply openapi.yaml:415]: PartnerNotFound describe not what actually is returned, I mentioned this when sent you screenshot

File: openapi.yaml [hunk 4]
```diff
@@ -389,6 +430,12 @@ components:
         application/json:
           schema:
             $ref: '#/components/schemas/GeneralError'
```
[the remark this answers is not in the document]
Comment (Hrnt125) [reply openapi.yaml:432]: I did not change this part, bit why for general 'BadRequest' we should return specific 'GeneralResellerNotFound'?

File: openapi.yaml [hunk 5]
```diff
@@ -723,6 +781,30 @@ components:
...
+          description: Error code
+          example: 'ERROR_CODE'
+        message:
+          type: string
+          description: User friendly description of above code
+          example: 'Error code explanation'
+        field:
+          type: string
+          description: Estimate error field
+          example: 'some field'
```
Comment (fruwe) [inline openapi.yaml:798]: NOTE: The API currently returns here `domainName`, but it should be `search`. Will need to get fixed in the future

**Review by fruwe — CHANGES_REQUESTED** [review]

File: openapi.yaml (hunk 1 above)
Comment (Hrnt125) [reply openapi.yaml:415]: --- Fixed ---

File: openapi.yaml (hunk 3 above)
Comment (Hrnt125) [reply openapi.yaml:50]: --- Fixed ---

File: openapi.yaml [hunk 6]
```diff
@@ -31,6 +31,25 @@ tags:
   - name: actions
     description: Create, sign and check status of blockchain actions
 paths:
+
+  /domains/:
+    parameters:
+      - $ref: '#/components/parameters/DomainNamesParameter'
+    get:
+      tags:
```
[the remark this answers is not in the document]
Comment (Hrnt125) [reply openapi.yaml:39]: --- Fixed ---

File: openapi.yaml [hunk 7]
```diff
@@ -723,6 +771,30 @@ components:
           type: number
           description: Status code of the response
           example: 400
+    DomainInvalidNameError:
```
Comment (fruwe) [inline openapi.yaml:774]: This seems to be no longer in use, let's remove this :)

**Review by fruwe — CHANGES_REQUESTED** [review]

File: openapi.yaml (hunk 7 above)
Comment (Hrnt125) [reply openapi.yaml:774]: I left it on purpose, like after MVP use it back, but np, I will remove an after add it if we need.

Comment (Hrnt125) [reply openapi.yaml]: --- Fixed ---

**Review by fruwe — APPROVED** [review]
LGTM

**Pull request merged**